src/mongo/commands.ts

```diff
@@ -38,7 +39,10 @@ export class MongoCommands {
 			if (!database) {
 				throw new Error('Please select a MongoDB database to run against by selecting it in the explorer and selecting the "Connect" context menu item');
 			}
-
+			if (command.errors) {
+				command.errors.sort((a, b) => a.position.character - b.position.character);
```

Comment:
The caller shouldn't be modifying data in the visitor.  You should either clone it first, or better do the sort in the visitor.